### PR TITLE
Allow Insecure Protocol to prevent warnings during build and upgrade to Scala 2.12.13 / sbt 1.3.0

### PR DESCRIPTION
**Pull request by jad-hamza**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -31,9 +31,9 @@ unmanagedJars in Compile += {
 }
 
 resolvers ++= Seq(
-  "Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots",
-  "Sonatype OSS Releases" at "https://oss.sonatype.org/content/repositories/releases",
-  "uuverifiers" at "http://logicrunch.research.it.uu.se/maven"
+  ("Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots").withAllowInsecureProtocol(true),
+  ("Sonatype OSS Releases" at "https://oss.sonatype.org/content/repositories/releases").withAllowInsecureProtocol(true),
```
Comment (samarion) [inline build.sbt:35]: Is this change necessary? These repos seem to be using 'https' which isn't an insecure protocol.

**Review by samarion — COMMENTED** [review]
You still need to update all the `.md` files under `src/main/doc` and replace "tut" with "mdoc".

Also, could you please check that adding a compilation error in the documentation correctly breaks compilation?

File: build.sbt (hunk 1 above)
Comment (jad-hamza) [reply build.sbt]: No you're right, it's fixed now thanks.

**Comment by jad-hamza** [comment]
> You still need to update all the `.md` files under `src/main/doc` and replace "tut" with "mdoc".
>
> Also, could you please check that adding a compilation error in the documentation correctly breaks compilation?

It all seems to compile properly now with `sbt mdoc` (I had to fix a few compilation errors).

I also added an extra case to silence a warning in normal Inox compilation:
https://github.com/epfl-lara/inox/pull/128/files#diff-e6b0bb85df4bd352bb54eb2b07207b020e5c04ed783e4df9982ccfcc83bf7d48R160

**Comment by jad-hamza** [comment]
There are many warnings related to documentation links but I think there were there before?

File: doc/interpolations.md [hunk 2]
```diff
@@ -44,26 +44,30 @@ import inox._
...
 ```scala
-scala> val tpe = t"Boolean"
-tpe: inox.trees.interpolator.trees.Type = Boolean
-
-scala> val expr = e"1 + 1 == 2"
-expr: inox.trees.interpolator.trees.Expr = 1 + 1 == 2
+val tpe = t"Boolean"
+// tpe: Type = Boolean
+val expr = e"1 + 1 == 2"
+// expr: Expr = Equals(
```
Comment (samarion) [inline doc/interpolations.md:56]: Do you know why pretty-printing is broken for expressions now?

**Comment by samarion** [comment]
> There are many warnings related to documentation links but I think there were there before?

Yeah I don't think that's new. I guess we should go through the source code and fix all the broken links some day...

File: doc/interpolations.md (hunk 2 above)
Comment (jad-hamza) [reply doc/interpolations.md:56]: Hmm no that's strange

**Comment by jad-hamza** [comment]
> Yeah I don't think that's new. I guess we should go through the source code and fix all the broken links some day...

The goal is to have the links work when browsing on GitHub?

**Comment by samarion** [comment]
> The goal is to have the links work when browsing on GitHub?

Aren't the warnings about broken links in the Scaladoc? If so, then we'd just get a bunch of broken links if we ever generated and published the Scaladoc.

If these are unrelated warnings, then I don't I know about them.

**Comment by jad-hamza** [comment]
Actually these seem to be false positives (see https://github.com/scalameta/mdoc/issues/94), and the links work well on GitHub, see for instance:
https://github.com/jad-hamza/inox/blob/allow-insecure/doc/trees.md

**Comment by samarion** [comment]
Hmm, I guess you can disable the link hygiene option: https://github.com/scalameta/mdoc/pull/129/files

**Comment by jad-hamza** [comment]
Done, thanks!

File: doc/interpolations.md (hunk 2 above)
Comment (samarion) [reply doc/interpolations.md:56]: It seems like we can use something like `mdoc:to-string` to use `toString()` instead of the weird custom printer: https://github.com/scalameta/mdoc/pull/148/files

Comment (jad-hamza) [reply doc/interpolations.md]: Thanks for looking this up! Should be good now

**Comment by jad-hamza** [comment]
Awesome, thanks!
